.svglintrc.mjs

```diff
@@ -148,6 +149,40 @@ const ignoreIcon = (linterName, path, $) => {
   iconIgnored[linterName][path] = iconName;
 };
 
+/**
+ * Cache for SVG icons.
+ *
+ * Used to avoid parsing the same data for each icon multiple times.
+ */
+class IconsCache {
+  static content = {
+    path: {},
+    segments: {},
+    bbox: {},
+  };
+
+  static getOrSetPath($icon, filepath) {
+    if (!this.content.path[filepath]) {
+      this.content.path[filepath] = $icon.find('path').attr('d');
+    }
+    return this.content.path[filepath];
+  }
+
+  static getOrSetSegments(iconPath) {
+    if (!this.content.segments[iconPath]) {
+      this.content.segments[iconPath] = parsePath(iconPath);
+    }
+    return this.content.segments[iconPath];
+  }
+
+  static getOrSetBbox(iconPath) {
+    if (!this.content.bbox[iconPath]) {
+      this.content.bbox[iconPath] = svgPathBbox(iconPath);
+    }
+    return this.content.bbox[iconPath];
+  }
+}
```

Comment:
I prefer to create a memoize function for caching results.

You could use https://github.com/sindresorhus/mem or https://github.com/sindresorhus/p-memoize directly or write a simple one.